proxy/http/HttpSM.cc

```diff
@@ -848,6 +848,12 @@ HttpSM::state_read_client_request_header(int event, void *data)
   case PARSE_RESULT_DONE:
     SMDebug("http", "[%" PRId64 "] done parsing client request header", sm_id);
 
+    if (!t_state.hdr_info.client_request.check_hdr_implements()) {
+      t_state.http_return_code = HTTP_STATUS_NOT_IMPLEMENTED;
```

Comment:
We should be responding with 415

https://tools.ietf.org/html/rfc7231#section-3.1.2.2
   An origin server MAY respond with a status code of 415 (Unsupported
   Media Type) if a representation in the request message has a content
   coding that is not acceptable.

Reply:
Dueling RFC's.  I'm moving the discussion back to issue #7590.